File: src/main/java/org/example/chessplatformbe/controller/dto/request/CreateReportDTO.java
package org.example.chessplatformbe.controller.dto.request;

import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class CreateReportDTO {
    private Integer updateId;

    private String description;

    @NotBlank(message = "Type is required")
    private String type;

    @NotNull(message = "Date is required")
    private LocalDateTime dateTime;

    @NotNull(message = "User is required")
    private Integer userId;
}

File: src/controller/dto/request/create_report_dto.rs
use chrono::NaiveDate;

pub struct CreateReportDTO {
    update_id: u32,
    description: String,
    r#type: String,
    date_time: NaiveDate,
    user_id: u32,
}

impl CreateReportDTO {
    pub fn new(
        update_id: u32,
        description: String,
        r#type: String,
        date_time: NaiveDate,
        user_id: u32,
    ) -> Self {
        Self {
            update_id,
            description,
            r#type,
            date_time,
            user_id,
        }
    }
}
